projects/graph/src/graph.py

```diff
@@ -1,9 +1,78 @@
-"""
-Simple graph implementation compatible with BokehGraph class.
-"""
+import random
 
 
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        """
+        Create an empty graph
+        """
+        self.vertices = {}
+    def add_vertex(self, vertex_id):
+        """
+        Add a vertex to the graph
+        """
+        self.vertices[vertex_id] = Vertex(vertex_id)
+    def add_edge(self, v1, v2):
+        """
+        Add an undirected edge to the graph
+        """
+        if v1 in self.vertices and v2 in self.vertices:
+            self.vertices[v1].edges.add(v2)
+            self.vertices[v2].edges.add(v1)
+        else:
+            raise IndexError("That vertex does not exist!")
+    def add_directed_edge(self, v1, v2):
+        """
+        Add a directed edge to the graph
+        """
+        if v1 in self.vertices:
+            self.vertices[v1].edges.add(v2)
+        else:
+            raise IndexError("That vertex does not exist!")
+    # def dft(self, starting_node=None, visited=None):
+    #     print('starting node', starting_node)
+    #     if visited is None:
+    #         visited = []
+    #     if starting_node is None: 
+    #         verticies = list(self.vertices.keys())
+    #         starting_node = verticies[0]
+    #         print('first starting node', starting_node)
+    #     visited.append(starting_node)
+        
+    #     for child in self.vertices[starting_node].edges:
+    #         if child not in visited:
+    #               self.dft(child, visited)
+    def dbft(self, popper=0):
+        verticies = list(self.vertices.keys())
+        q = [verticies[0]]
```

Comment:
This is great.  You could pass in a value to your method and use that as your starting point.